### ISSUE TITLE
(Android) PPSSPP sees two different controller inputs as "Pad 1" 

### ISSUE THREAD

**Issue by Angstdiscord** [hrydgard/ppsspp#18602]
### Game or games this happens in

N/A

### What area of the game / PPSSPP

When using my Odin 2 the built in controls and the external bluetooth controller I'm using (Dualsense Edge) are both seen as the same controller (Pad 1) This causes issues because I cannot map the controls correctly. 

For example, the built in controls are a nintendo layout, so I try to configure them to match PS
A as Circle (Pad 1)
B as Cross (Pad 1)
X as Triangle (Pad 1)
Y as Square. (Pad 1)

This is fine, but then when I try to map the Dualsense to its correctly layout it shows
B as Circle (Pad 1)
A as Cross (Pad 1)
Y as Triangle (Pad 1)
X as Square (Pad 1)

These are two separate controllers but since it shows them as a single controller, it maps two face buttons to each input.

Auto Configure recognizes both controllers (It sees both the dualsense and odin controller) but since they are both pad1, its causing controller issues.

### What should happen

PPSSP should recognize both controllers as seperate devices with seperate inputs. This should allow two controllers to be configured to work with PPSSPP at the same time.

### Logs

![IMG_20231221_172739](https://github.com/hrydgard/ppsspp/assets/81662744/a133e549-1037-46f2-8610-6c09577e776f)

### Platform

Android

### Mobile device model or graphics card (GPU)

Odin 2

### PPSSPP version affected

v1.16.6-657-g986bf5ed9c

### Last working version

N/A

### Graphics backend (3D API)

Vulkan

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [X] Try without any cheats and without loading any save states.
- [X] Include logs or screenshots of issue.

**Comment by Angstdiscord** [comment]
Any advice? I know you guys are busy

**Comment by AlySaberSadek** [comment]
Hey, I'm running into the same problem. Did you figure out how to fix it?

**Comment by numb3rb0y** [comment]
Hi, I'm also having the same issue with PPSSPP and the built-in controls on an Anbernic RG556 and a bluetooth-connected DualShock 4, would appreciate any workaround/fix.

**Comment by hrydgard** [comment]
Yeah, this is a limitation of the current controller mapping system, we don't store any kind of controller ID with the mapping. Still a problem, sorry :(
